brayns/Brayns.cpp

```diff
@@ -178,24 +189,21 @@ struct Brayns::Impl
 
     void render()
     {
-        Scene& scene = _engine->getScene();
-        Camera& camera = _engine->getCamera();
+        if( _engine->isDirty( ))
+            createEngine();
+
         const Vector2ui windowSize = _parametersManager->getApplicationParameters().getWindowSize();
+
         _engine->reshape( windowSize );
+        _engine->preRender();
 
+        _extensionPluginFactory->execute( *_engine );
+
+        _engine->reshape( windowSize );
         _engine->preRender();
 
```

Comment:
Maybe check the engine pointer and only apply the reshape + preRender if it was modified.